File: WorkWithFile.java
package Second_sem.lab5;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.internal.LinkedTreeMap;

import java.io.*;
import java.util.LinkedList;
import java.util.Scanner;

public class WorkWithFile {

    public static String readFromFile(String pathToFile) {
        try {
            return new Scanner(new File(pathToFile)).useDelimiter("\\Z").next();
        } catch (FileNotFoundException e) {
            throw new RuntimeException(e);
        }
    }

    public static LinkedList<LinkedTreeMap> convertJSONtoLinkedList(String txt){
        return new Gson().fromJson(txt, LinkedList.class);
    }



    public static void writeInFile(String pathToFile, LinkedList<LinkedTreeMap> data){
        Gson gson = new GsonBuilder().serializeNulls().setPrettyPrinting().create();
        String jsonString = gson.toJson(data);
        try (BufferedOutputStream bos = new BufferedOutputStream(new FileOutputStream(pathToFile))) {
            bos.write(jsonString.getBytes());
        } catch (IOException|RuntimeException e) {
            throw new RuntimeException(e);
        }
    }
}

File: Kotlin/HelpingOrFormatingClasses/WorkWithFile.kt
package Second_sem.lab5.Kotlin.HelpingOrFormatingClasses

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.google.gson.internal.LinkedTreeMap
import java.io.*
import java.util.*


fun readFromFile(pathToFile: String): String {
    var txt = ""
    try {
        txt = Scanner( File(pathToFile)).useDelimiter("\\Z").next()
    } catch (e: FileNotFoundException) {
        //todo add print by class
        println("File not found")
    }
    return txt
}


fun convertJSONtoLinkedList(txt: String) : LinkedList<LinkedTreeMap<String, Any?>> {
    return Gson().fromJson(txt, LinkedList::class.java) as LinkedList<LinkedTreeMap<String, Any?>>
}


fun writeInJSONFile(pathToFile: String, data: LinkedList<*>) {
    val gson = GsonBuilder().serializeNulls().setPrettyPrinting().create()
    val jsonString = gson.toJson(data)
    try  {
       val bos = BufferedOutputStream(FileOutputStream(pathToFile))
        bos.write(jsonString.toByteArray())
    }
    catch (e: IOException){
        throw IOException(e)
    }
    catch (e: RuntimeException){
        throw RuntimeException(e)
    }
}


fun writeInTxtFile(pathToFile: String, data: String){
    try {
        val fileWriter = FileWriter(pathToFile)
        fileWriter.write(data)
        fileWriter.flush()
        fileWriter.close()
    }
    catch (e: IOException){
        throw IOException(e)
    }
}